cmd/snap/cmd_warnings.go

```diff
@@ -0,0 +1,210 @@
+
+func init() {
+	addCommand("warnings", shortWarningsHelp, longWarningsHelp, func() flags.Commander { return &cmdWarnings{} }, timeDescs.also(map[string]string{
+		"all":     i18n.G("Show all warnings"),
+		"verbose": i18n.G("Show more information"),
+	}), nil)
+	addCommand("okay", shortOkayHelp, longOkayHelp, func() flags.Commander { return &cmdOkay{} }, nil, nil)
+}
+
+func (cmd *cmdWarnings) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	warnings, err := cmd.client.Warnings(client.WarningsOptions{All: cmd.All})
+	if err != nil {
+		return err
+	}
+	if len(warnings) == 0 {
+		fmt.Fprintln(Stderr, i18n.G("No new warnings."))
+		return nil
+	}
+
+	if err := writeWarningTimestamp(time.Now()); err != nil {
+		return err
+	}
+
+	w := tabWriter()
+	if cmd.Verbose {
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n",
+			i18n.G("First occurrence"),
+			i18n.G("Last occurrence"),
+			i18n.G("Expires after"),
+			i18n.G("Acknowledged"),
+			i18n.G("Repeats after"),
+			i18n.G("Warning"))
+		for _, warning := range warnings {
+			lastShown := "-"
+			if warning.LastShown != nil && !warning.LastShown.IsZero() {
+				lastShown = cmd.fmtTime(*warning.LastShown)
+			}
+			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\n",
+				cmd.fmtTime(warning.FirstAdded),
+				cmd.fmtTime(warning.LastAdded),
+				warning.ExpireAfter,
+				lastShown,
+				warning.RepeatAfter,
+				warning.Message)
+		}
+	} else {
+		fmt.Fprintf(w, "%s\t%s\n", i18n.G("Last occurrence"), i18n.G("Warning"))
+		for _, warning := range warnings {
+			fmt.Fprintf(w, "%s\t%s\n", cmd.fmtTime(warning.LastAdded), warning.Message)
+		}
+	}
+	w.Flush()
+
+	return nil
+}
+
+func (cmd *cmdOkay) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	last, err := lastWarningTimestamp()
+	if err != nil {
+		return fmt.Errorf("no client-side warning timestamp found: %v", err)
+	}
+
+	return cmd.client.Okay(last)
+}
+
+const warnFileEnvKey = "SNAPD_LAST_WARNING_TIMESTAMP_FILENAME"
+
+func warnFilename(homedir string) string {
+	if fn := os.Getenv(warnFileEnvKey); fn != "" {
+		return fn
+	}
+
+	return filepath.Join(homedir, ".snap", "warning")
+}
+
+type clientWarningData struct {
+	Timestamp time.Time `json:"timestamp"`
+}
+
+func writeWarningTimestamp(t time.Time) error {
+	user, err := osutil.RealUser()
+	if err != nil {
+		return err
+	}
+	uid, gid, err := osutil.UidGid(user)
+	if err != nil {
+		return err
+	}
+
+	aw, err := osutil.NewAtomicFile(warnFilename(user.HomeDir), 0600, 0, uid, gid)
+	if err != nil {
+		return err
+	}
+	// Cancel once Committed is a NOP :-)
+	defer aw.Cancel()
+
+	enc := json.NewEncoder(aw)
+	if err := enc.Encode(clientWarningData{Timestamp: t}); err != nil {
+		return err
+	}
+
+	return aw.Commit()
+}
+
+func lastWarningTimestamp() (time.Time, error) {
+	user, err := osutil.RealUser()
+	if err != nil {
+		return time.Time{}, fmt.Errorf("cannot determine real user: %v", err)
+	}
+	f, err := os.Open(warnFilename(user.HomeDir))
```

Comment:
A minimalistic solution would be to use an empty file and just look at the modification timestamp like systemd does with timer files.